internal/controller/providers/pg/applier.go

```diff
@@ -136,6 +141,9 @@ func (p *applier) Engine() error {
 	if err := p.updatePGConfig(pg, database); err != nil {
 		return errors.Join(err, errors.New("could not update PG config"))
 	}
+
+	pg.Spec.InstanceSets = defaultSpec().InstanceSets
+
```

Comment:
why is it required? We already set the default spec with InstanceSets to `p.PerconaPGCluster.Spec`

Reply:
Where? Do you mean in `ResetDefaults()`?